### ISSUE TITLE
Add maven-styled semantic versioning control

### ISSUE THREAD

**Issue by fralalonde** [mojohaus/versions-maven-plugin#144]
Because the release plugin is somewhat rigid in its operation, the version plugin becomes the fallback tool for custom release scripts and profiles. Unfortunately, simple and safe version transformations are often harder than they should be. 

By adopting a standard [semver-ish](https://cwiki.apache.org/confluence/display/MAVEN/Version+number+policy) format matching the existing maven version convention, the version plugin could allow manipulating individual component of the version string using targeted parameters. For example, incrementing the minor version, resetting the patch version to 0, setting / unsetting a branch qualifier without affecting the snapshot version state.

A maven-semver format of &lt;major&gt;[.&lt;minor&gt;[.&lt;patch&gt;]][&lt;qualifier&gt;][-SNAPSHOT] would cover a lot of existing version schemes.

**Comment by khmarbaise** [comment]
You can already use the [build-helper-maven-plugin](http://www.mojohaus.org/build-helper-maven-plugin/parse-version-mojo.html) in combination with versions-maven-plugin like the following:
```
mvn build-helper:parse-version versions:set \
     -DnewVersion=\${parsedVersion.majorVersion}.\${parsedVersion.nextMinorVersion}.0 \
     versions:commit
```
May be it would be a good idea to integrate the `parse-version` into the versions-maven-plugin as well...

**Comment by fralalonde** [comment]
I am sneaky :) and already had a branch for the proposal, which I've submitted a PR for : https://github.com/mojohaus/versions-maven-plugin/pull/145
It's a basis for discussion - I wrote this over the last 24 hours. I've reused the `SetMojo` code where possible by introducing a shared parent class `AbstractModuleVersionUpdaterMojo`. I've tested it manually and it works.

The build-helper method you suggest is also nice because of the format flexibility it allows and also  because it is made entirely of existing parts. The resulting syntax is a bit cumbersome, but this can be lessened with a good comment. I was not aware of that feature of build-helper, some of these things should simply be in core maven IMO...
